tap_bitbucket/gitlocal.py

```diff
@@ -91,13 +104,22 @@ def parseDiffLines(lines, shouldEncrypt=False, hmacToken=None):
       if line[0] == '@':
         # Note: this line may have context at the end, which is okay and part of the git difff
         # format.
-        curChange['patch'].append(hashPatchLine(line, hmacToken) if shouldEncrypt else line)
+        if curChange['patch_lines'] < MAX_DIFF_LINES and curChange['patch_length'] < MAX_PATCH_SIZE:
+          patchLine = hashPatchLine(line, hmacToken) if shouldEncrypt else line
+          curChange['patch'].append(patchLine)
+          curChange['patch_lines'] += 1
+          curChange['patch_length'] += len(patchLine)
       else:
         if line[0] == '-':
           curChange['deletions'] += 1
         elif line[0] == '+':
           curChange['additions'] += 1
-        curChange['patch'].append(hashPatchLine(line, hmacToken) if shouldEncrypt else line)
+        
+        if curChange['patch_lines'] < MAX_DIFF_LINES and curChange['patch_length'] < MAX_PATCH_SIZE:
```

Comment:
If we go past either of these thresholds, we should completely clear the patch lines and mark the file as a large file.